modules/oe_translation_poetry/src/Event/PoetryRequestTypeEvent.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_translation_poetry\Event;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\oe_translation_poetry\PoetryRequestType;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Event class for determining what kind of Poetry request can be made.
+ */
+class PoetryRequestTypeEvent extends Event {
+
+  const EVENT = 'oe_translation_poetry.request_type_event';
+
+  /**
+   * The entity being translated.
+   *
+   * @var \Drupal\Core\Entity\ContentEntityInterface
+   */
+  protected $entity;
+
+  /**
+   * The request type constant.
+   *
+   * @var \Drupal\oe_translation_poetry\PoetryRequestType
+   */
+  protected $requestType;
+
+  /**
+   * The job information of the current request.
+   *
+   * @var array
+   */
+  protected $jobInfo;
+
+  /**
+   * PoetryRequestTypeEvent constructor.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The entity being translated.
+   * @param \Drupal\oe_translation_poetry\PoetryRequestType $requestType
+   *   The request type constant.
+   * @param object|null $jobInfo
```

Comment:
I would prefer to have here either an integer ($job_info->tjid) or already the JobInterface $job object.

Like this the $job_info argument is not clear what should be.

I put another change request similar to this one after.
But here it's even more important because it's impacting code in other components.

Reply:
Mmm, there we need more than just the job object. We need also the Job Item ID..so I will document better the object instead.